wandb/sdk/launch/agent/agent.py

```diff
@@ -359,19 +353,20 @@ def run_job(
 
         # Abort if this job attempts to override secure mode
         self._assert_secure(launch_spec)
-
-        self._pool.apply_async(
-            self.thread_run_job,
-            (
+        job_tracker = JobAndRunStatusTracker(job["runQueueItemId"], queue, file_saver)
+        t = threading.Thread(
+            target=self.thread_run_job,
+            args=(
                 launch_spec,
                 job,
                 self.default_config,
                 self._api,
-                queue,
-                file_saver,
+                job_tracker,
             ),
         )
 
+        t.start()
```

Comment:
If I run the agent outside of a cluster and I ctrl-c it while these threads are running I think they will continue to run. If we set `daemon=True` then they will automatically exit when all non daemon threads (aka the main agent thread) have finished